### DISCUSSION TITLE
Unequal column widths for Grid component

### DISCUSSION THREAD

**Discussion by AleenaSA** [twilio-labs/paste#3436] [Q&A] [answered]
This is the sample snippet I've been referring to https://paste.twilio.design/components/grid#stacked-columns-in-a-vertical-layout

<img width="400" alt="image" src="https://github.com/twilio-labs/paste/assets/29952750/6f5635c7-3c7b-414e-be95-760b55e086a1">

While the `offset` property seems to work, `span` property doesn't. There's a use-case for us wherein we require one of the cards to be of different width. This could work with, for e.g., a nested grid. Is there a non-hacky solution for the same?

For context:
we need something like this, the bottom card should be half the width. And on smaller screen, it should be displayed last
<img width="400" alt="image" src="https://github.com/twilio-labs/paste/assets/29952750/03a25603-3876-4285-9867-7e63d1d94389">

**Comment by TheSisb** [comment]
Something like this? https://codesandbox.io/s/paste-starter-template-forked-dccf92?file=/src/index.tsx

The key is to keep the total `span` and `offset` to equal 12 columns total, and using responsive values (the array index represents each breakpoint).

**Reply by AleenaSA** [reply]
hey, no this is not what we're expecting. In the vertical grid, the cards are required to be of different widths
i.e. instead of 
<img width="400" alt="image" src="https://github.com/twilio-labs/paste/assets/29952750/47299913-faaf-4712-9736-6ef162e7e3fc">

 something like this 
<img width="400" alt="image" src="https://github.com/twilio-labs/paste/assets/29952750/c359c48c-ee8d-4fff-be6e-078dbd51c9e6">


The snippet, with nested grid as a workaround: https://codesandbox.io/s/paste-starter-template-forked-pcgcqg

**Reply by TheSisb** [reply, marked as answer]
Oh I see. Yeah it's strange, it only works if you provide an offset like this: https://codesandbox.io/s/paste-starter-template-forked-67r4j2?file=/src/index.tsx

I'll make a ticket to investigate further, but passing a 0 offset should unblock you for now
